File: java/javadpbo/src/com/sem4dpbo/week1/methode/Titik.java
package com.sem4dpbo.week1.methode;

public class Titik {
    private int x;          //koodinat x
    private int y;          //koordinat y

    //constructor 1
    Titik(){
        x = 0;
        y = 0;
    }

    //constructor 2
    Titik(int xp, int yp){
        this.x = xp;
        this.y = yp;
    }

    //fungsi set
    public void setX(int xp){
        x = xp;
    }

    public void setY(int yp){
        y = yp;
    }

    //fungsi get
    int getX(){
        return x;
    }

    int getY(){
        return y;
    }
}

File: cpp/9th_Week/template/implementasi1/Titik.cpp
#include "Titik.h"
using namespace std;

// template <class T>
// Titik<T>::Titik(){
//     // konstruktor
//     Titik<T>::x = 0;
//     Titik<T>::y = 0;
// }

template <class T>
Titik<T>::Titik(int xp, int yp){
    // konstruktor
    Titik<T>::x = xp;
    Titik<T>::y = yp;
}

template <class T>
int Titik<T>::getX(){
    return Titik<T>::x;
}

template <class T>
void Titik<T>::setX(int xp){
    Titik<T>::x = xp;
}

template <class T>
int Titik<T>::getY(){
    return Titik<T>::y;
}

template <class T>
void Titik<T>::setY(int yp){
    Titik<T>::y = yp;
}

template <class T>
Titik<T>::~Titik(){}
